qiskit/circuit/commutation_checker.py

```diff
@@ -103,6 +119,20 @@ def commute(
         Returns:
             bool: whether two operations commute.
         """
+        # The rotation gates commute like their respective generators.
+        if (
+            op1.name in _rotation_gates
+            and getattr(op1, "is_parameterized", False)
+            and op1.is_parameterized()
```

Comment:
I thought we could reduce rotation gates to Paulis independent of whether they have free parameters or not. If they have concrete parameters, the current code computes the matrix representation afaik, but if we'd check the underlying Pauli instead we could leveraged the cached table which should be faster 🙂

Reply:
I agree with that. We would need to be careful that we are not missing some cases in which the general rotation does not commute but the paramterized one does. I am thinking mainly of theta%2*np.pi is close to 0.